File: CS50 2019/C/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    int coins;
    do
    {
        float f = get_float("Change: ");
        coins = round(f * 100);
    }
    while(coins <= 0);
    
    int quarters = coins / 25;
    int dimes = (coins % 25) / 10;
    int nickels = ((coins % 25) % 10) / 5;
    int pennies = ((coins % 25) % 10) % 5;

    int count = quarters + dimes + nickels + pennies;

    printf("%d\n", count);
}

File: CS50 2019/Python/pset6/cash.py
from cs50 import get_float
from math import floor

while True:
    f = get_float("Change owed: ")
    fl = floor(f * 100)
    if f > 0:
        break
    

quarters = fl // 25
dimes = (fl % 25) // 10
nickels = ((fl % 25) % 10) // 5
pennies = ((fl % 25) % 10) % 5

count = quarters + dimes + nickels + pennies

print(f"{count}")